src/common/map/MapDirective.js

```diff
@@ -53,15 +53,44 @@
 
             scope.$watch('layers', function(layers) {
               if (layers && map) {
-                var layerLength = map.getLayers().getLength();
-                for (var i = 1; i < layerLength; i++) {
-                  map.removeLayer(map.getLayers().getArray()[1]);
+                deleteNoCartLayers();
```

Comment:
why do you remove all layers and then in addAllLayers you remove them once again?